### ISSUE TITLE
I installed from the market place and didn't get a project template added

### ISSUE THREAD

**Issue by cedral** [Microsoft/vs-boost-unit-test-adapter#2]
I have rebooted and still no dice. The project template used to be Visual C++ -> Test -> Boost Unit Test Project. I have like > 300 tests I need to port to visual studio 2017 and I really need this to work.
<img width="581" alt="screen shot 2017-08-15 at 4 16 09 pm" src="https://user-images.githubusercontent.com/3732140/29334396-42b6932e-81d5-11e7-84bb-100ad6e705d1.png">
<img width="1105" alt="screen shot 2017-08-15 at 4 15 33 pm" src="https://user-images.githubusercontent.com/3732140/29334397-42c6cc12-81d5-11e7-9f71-32d3feeddcb0.png">
<img width="533" alt="screen shot 2017-08-15 at 4 14 21 pm" src="https://user-images.githubusercontent.com/3732140/29334411-48bb4300-81d5-11e7-950c-ada9ebba6a13.png">
<img width="502" alt="screen shot 2017-08-15 at 4 14 01 pm" src="https://user-images.githubusercontent.com/3732140/29334398-42cb6128-81d5-11e7-9172-d4af6201e284.png">

**Comment by LukaszMendakiewicz** [comment]
Yes, this fork has project templates removed as mentioned in [the release notes](https://github.com/Microsoft/vs-boost-unit-test-adapter/blob/dev15/BoostTestPlugin/Release%20Notes.txt#L17). We are planning to add them back in, but we do not have a specific timeframe to offer.

As an alternative, you should be able to manually create Boost.Test project from scratch, acquiring Boost.Test from the official Boost repository, through [vcpkg](https://github.com/Microsoft/vcpkg), or other distributions.

If you have Boost.Test project created in Visual Studio 2015 you want to port to Visual Studio 2017, you should be able to open those project in the latter version and upgrade them (it should be offered automatically, or in the project context menu in the Solution Explorer).

**Comment by guwirth** [comment]
@cedral you can also have a look to https://github.com/etas/vs-boost-unit-test-adapter/tree/master/BoostTestPlugin for the templates and adapt it for VS2017. Maybe @LukaszMendakiewicz can do this/help you in this repository?

**Comment by cedral** [comment]
I think it is odd that was not mentioned in the blog post where the other changes in those release notes were listed. I just wasted half a day.
I have existing test projects. I can't really use them for this. visual studio 2017 has significant new features. My old tests won't work with /std:c++latest but the new tests need to be run using that since they use std::string_view. They actually need to be ported which is why I want a new project.
When do you think you might have this done. If it is going to be 2 weeks I can just wait. If it will be months then I need to try and copy and rename my existing projects which will probably require hacking the project files. sigh. I wish you hadn't talked the previous maintainer into abandoning this until you were really ready to take it over. 
This statement require clarification before I can make heads or tails of it: "you should be able to manually create Boost.Test project from scratch, acquiring Boost.Test from the official Boost repository, through vcpkg, or other distributions."
Boost.Test knows nothing about visual studio projects. It is intended to be used with BJAM from the command line. If I want to run tests that way I can just stop using visual studio.

**Comment by LukaszMendakiewicz** [comment]
Fair point about the blog post and apologies for your trouble. I will follow up with @AugP if he can update the post so it will not trip the next person.

Please note that project files are always upgradeable between Visual Studio version. So in general you should be able to take any project file supported by Visual Studio 2015, open it in Visual Studio 2017, update it to the new toolset version, and use any new compiler/library/IDE features (including /std:c++latest).

We definitely will not have project templates available in the next 2 weeks if that helps you make a decision on how to proceed further.

For the record, I don't think that "talked the previous maintainter into abanodning this" is accurately describing the relationship we have with @guwirth's team. This fork was started to help them support Visual Studio 2017 and it was the upstream project's decision to let us take the reins. The adapter is still fully open source under a license allowing third-party modifications and we welcome both community contributions and forks.

The steps to set-up a Boost.Test project manually from scratch are not that far off from the experience using the project template in the Boost Unit Test Adapter in Visual Studio 2015:
1) As was the case with the Boost Unit Test Adapter in Visual Studio 2015, you need to acquire Boost first through your preferred mechanism. It could be built from source or downloaded pre-built.
2) In Visual Studio 2017, create new "Windows Console Application" project.
3) Provide Additional Include Directories and Additional Library Directories to the project, as described in [the original project's wiki](https://github.com/etas/vs-boost-unit-test-adapter/wiki/User-Manual#usage).
4) Replace `int main() { ... }` in the created .cpp file with the following code:
```
#include <boost/test/unit_test.hpp>

#define BOOST_TEST_MODULE BoostUnitTest1

BOOST_AUTO_TEST_CASE(BoostUnitTest)
{
    // TODO: Your test code here
}
```
5) Build the project.
6) Open Test Explorer from Test > Windows menu.

After a short discovery process the test should be listed in the Test Explorer window. From now on the experience should be the same you have had in Visual Studio 2015 using the original adapter project. Please note that most of the information present in the excellent [User Manual](https://github.com/etas/vs-boost-unit-test-adapter/wiki/User-Manual) from the upstream project is still applicable to this fork.

**Comment by cedral** [comment]
Thanks. That is easy. I can totally do that. We already have boost and I already know how to make an exe test module like that. I just assumed that tests getting listed into the test runner required magic in the project file.

**Comment by LukaszMendakiewicz** [comment]
Glad to hear that!

We will keep this issue active to track the request to add the project templates to the adapter.

**Comment by guwirth** [comment]
> For the record, I don't think that "talked the previous maintainer into abandoning this" is accurately describing the relationship we have with @guwirth's team. This fork was started to help them support Visual Studio 2017 and it was the upstream project's decision to let us take the reins. The adapter is still fully open source under a license allowing third-party modifications and we welcome both community contributions and forks.

Also for the record: agree wholeheartedly and like to ask all to contribute to this repository in future.
